src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosCoinviewRule.cs

```diff
@@ -162,7 +162,7 @@ private void CheckAndComputeStake(RuleContext context)
                 // Compute stake modifier.
                 ChainedHeader prevChainedHeader = chainedHeader.Previous;
                 BlockStake blockStakePrev = prevChainedHeader == null ? null : this.stakeChain.Get(prevChainedHeader.HashBlock);
-                blockStake.StakeModifierV2 = this.stakeValidator.ComputeStakeModifierV2(prevChainedHeader, blockStakePrev, blockStake.IsProofOfWork() ? chainedHeader.HashBlock : blockStake.PrevoutStake.Hash);
+                blockStake.StakeModifierV2 = this.stakeValidator.ComputeStakeModifierV2(prevChainedHeader, blockStakePrev?.StakeModifierV2, blockStake.IsProofOfWork() ? chainedHeader.HashBlock : blockStake.PrevoutStake.Hash);
```

Comment:
Its probably not right to do this `blockStakePrev?.StakeModifierV2` we will fail and not know why.
At this stage we expect to have stake modifier and if not we need to throw.

Reply:
throw here or null-check into ComputeStakeModifierV2 to see if prevStakeModifier is null?
previous code where throwing into ComputeStakeModifierV2 on the line `serializer.ReadWrite(blockStakePrev.StakeModifierV2);` so maybe I should put a Guard notnull there to have the same behavior

Reply:
yes put a guard there